Add toast message when no user with common interests is found using filter

Yash-Raj-Singh opened the pull request:
### Description

<!--- Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change. --->
Added Toast for the situation when no user with common interests are not found.

Fixes #864 

### Type of Change:
<!--- **Delete irrelevant options.** --->

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)


### How Has This Been Tested?
Manually tested on my smartphone.
- Device - Samsung M21
- OS - Android 11


### Checklist:
<!--- **Delete irrelevant options.** --->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] Any dependent changes have been merged


**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### Video : 
https://user-images.githubusercontent.com/57872757/129070258-7620b66e-7ae7-42f0-8358-e98f632dea8a.mp4

Yash-Raj-Singh committed:
docs: add Logout Screen Page Screenshot to README.md (#1150)

Yash-Raj-Singh committed:
Add toast message when no user with common interests is found using filter

Yash-Raj-Singh commented:
@epicadk I have a question, What is the limit of item shown per page by a recyclerview ?

epicadk commented:
> @epicadk I have a question, What is the limit of item shown per page by a recyclerview ?

Not sure if this was changes but I think the default limit is 20? cc @anitab-org/mentorship-maintainers

Yash-Raj-Singh commented:
> > @epicadk I have a question, What is the limit of item shown per page by a recyclerview ?
> 
> Not sure if this was changes but I think the default limit is 20? cc @anitab-org/mentorship-maintainers

I did change it to 100 purposefully, cause otherwise it was searching only in those initial 20 items shown by recyclerview. I tried to extend it to 250 or 500 but figured out that 100 was a limit. Basically in some manner it is helping us to search in a broad data.
What do you think? @epicadk

epicadk commented on app/src/main/java/org/systers/mentorship/utils/Constants.kt:
```diff
@@ -2,7 +2,7 @@ package org.systers.mentorship.utils
 object Constants {
 
-    const val ITEMS_PER_PAGE = 20
+    const val ITEMS_PER_PAGE = 100
```
20 is fine.

Yash-Raj-Singh committed:
Changed the Item per page limit from 100 back to 20.

epicadk commented:
Please solve the merge conflicts. The rest of it seems good. 👍🏽

Yash-Raj-Singh commented:
> Please solve the merge conflicts. The rest of it seems good. 👍🏽

@epicadk Can you please tell me how to remove this conflict, cause I didn't understood well what to do here?

devkapilbansal commented:
> > Please solve the merge conflicts. The rest of it seems good. 👍🏽
> 
> @epicadk Can you please tell me how to remove this conflict, cause I didn't understood well what to do here?

@Yash-Raj-Singh you need to fetch changes from develop branch of this repository and merge/rebase your branch. Let us know if you need any help

Yash-Raj-Singh commented:
> > > Please solve the merge conflicts. The rest of it seems good. 👍🏽
> > 
> > 
> > @epicadk Can you please tell me how to remove this conflict, cause I didn't understood well what to do here?
> 
> @Yash-Raj-Singh you need to fetch changes from develop branch of this repository and merge/rebase your branch. Let us know if you need any help

Thanks for quick assistance @devkapilbansal. So if I fetch the new updated develop branch and perform the changes there, won't it create a new PR?

devkapilbansal commented:
> > > > Please solve the merge conflicts. The rest of it seems good. 👍🏽
> > > 
> > > 
> > > @epicadk Can you please tell me how to remove this conflict, cause I didn't understood well what to do here?
> > 
> > 
> > @Yash-Raj-Singh you need to fetch changes from develop branch of this repository and merge/rebase your branch. Let us know if you need any help
> 
> Thanks for quick assistance @devkapilbansal. So if I fetch the new updated develop branch and perform the changes there, won't it create a new PR?

Nah, you need to rebase this branch and push the changes.
This is how you should proceed:
```
# fetch the develop branch
git remote add upstream https://github.com/anitab-org/mentorship-android.git
git fetch upstream develop
# checkout on your branch
git checkout Toast_after_no_common_interest_found
# perform a rebase
git rebase upstream develop
# there will be some conflicts shown in the file mentioned above, i.e. `MembersFragment.kt`. Edit that file manually and do the changes.
# once done add your changes and continue
git add `app/src/main/java/org/systers/mentorship/view/fragments/MembersFragment.kt`
git rebase --continue
# force push your changes
git push -f
```

**Note**:- You can use merge instead of rebase too. Both will make the similar changes but they differ in how they work.
Also, I will suggest get some hands on experience with git, https://learngitbranching.js.org/

Yash-Raj-Singh commented:
> > > > > Please solve the merge conflicts. The rest of it seems good. 👍🏽
> > > > 
> > > > 
> > > > @epicadk Can you please tell me how to remove this conflict, cause I didn't understood well what to do here?
> > > 
> > > 
> > > @Yash-Raj-Singh you need to fetch changes from develop branch of this repository and merge/rebase your branch. Let us know if you need any help
> > 
> > 
> > Thanks for quick assistance @devkapilbansal. So if I fetch the new updated develop branch and perform the changes there, won't it create a new PR?
> 
> Nah, you need to rebase this branch and push the changes.
> This is how you should proceed:
> 
> ```
> # fetch the develop branch
> git remote add upstream https://github.com/anitab-org/mentorship-android.git
> git fetch upstream develop
> # checkout on your branch
> git checkout Toast_after_no_common_interest_found
> # perform a rebase
> git rebase upstream develop
> # there will be some conflicts shown in the file mentioned above, i.e. `MembersFragment.kt`. Edit that file manually and do the changes.
> # once done add your changes and continue
> git add `app/src/main/java/org/systers/mentorship/view/fragments/MembersFragment.kt`
> git rebase --continue
> # force push your changes
> git push -f
> ```
> 
> **Note**:- You can use merge instead of rebase too. Both will make the similar changes but they differ in how they work.
> Also, I will suggest get some hands on experience with git, https://learngitbranching.js.org/

@devkapilbansal Thanks, gonna do this asap.

Yash-Raj-Singh committed:
feat: add toast when no user is filtered by location (#1149)

# Conflicts:
#	app/src/main/java/org/systers/mentorship/view/fragments/MembersFragment.kt
#resolved conflict

Yash-Raj-Singh committed:
Resolved Conflict

Yash-Raj-Singh commented:
@devkapilbansal I guess I messed up something, I am not getting it. And git rebase --continue was not working even after manually editing the file and performing git add . I have forked this repo so there an option to fetch upstream in github, I did that and then tried fixing it. Now even the resolve conflict option is showing something else.
Can you check my last commit ?

Yash-Raj-Singh committed:
Merge branch 'develop' into Toast_after_no_common_interest_found

Yash-Raj-Singh commented:
@devkapilbansal @vj-codes Solved the merge conflict. Please have a look. :)

Yash-Raj-Singh commented:
@isabelcosta Can you please review the code? Earlier it was having some merge conflicts, now all seems to be good. Thanks :)

Yash-Raj-Singh commented:
@devkapilbansal Review needed, conflict has been resolved. 
Please have a look.

epicadk commented on app/src/main/java/org/systers/mentorship/view/fragments/MembersFragment.kt:
```diff
@@ -117,13 +117,15 @@ class MembersFragment : BaseFragment() {
                         if (!isRecyclerView) {
                             rvMembers.apply {
                                 layoutManager = LinearLayoutManager(context)
+
```
don't need this change.

Yash-Raj-Singh replied:
I actually didn't get it, Are you asking to keep the code writing format/syntax to be as earlier (new line, extra spaces) or something else?
In all three comments?

epicadk replied:
yes

Yash-Raj-Singh replied:
got it, On it :)

epicadk commented on app/src/main/java/org/systers/mentorship/view/fragments/MembersFragment.kt:
```diff
@@ -117,13 +117,15 @@ class MembersFragment : BaseFragment() {
 
                                 val dividerItemDecoration = DividerItemDecoration(
-                                        this.context, DividerItemDecoration.VERTICAL)
+                                    this.context, DividerItemDecoration.VERTICAL
```
same

epicadk commented on app/src/main/java/org/systers/mentorship/view/fragments/MembersFragment.kt:
```diff
@@ -132,12 +134,37 @@ class MembersFragment : BaseFragment() {
 
                                 val hasUsersWithLocation = membersViewModel.userList.any {
-                                    (it.location)?.contains(filterMap["location"]!!, ignoreCase = true) == true
+
```
ditto

Yash-Raj-Singh committed:
Updated with syntax/writing format fixes.

Yash-Raj-Singh committed:
Updated with syntax/writing format fixes. :)

Yash-Raj-Singh commented:
@epicadk Done bro 👍🏽  Check this out :)

vj-codes reviewed (approved):
LGTM 🎉

Yash-Raj-Singh commented:
@isabelcosta @epicadk Can you please review this PR, its been a long time? 😅
